### PR TITLE
Refresh: Add contracts, remove price calculator

### PR DESCRIPTION
**Pull request by matthijskrul**
## Summary of the changes / Why this is an improvement
- Adds information for Azure and AWS CrateDB Cloud Contracts
- Removes documentation and links for the now deprecated price calculator
- Small changes

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Review by lukas-bernert — APPROVED** [review]

File: docs/subscription-plans.rst [hunk 1]
```diff
@@ -63,6 +63,47 @@ optimized**.
...
+To sign a CrateDB Cloud Contract via Microsoft Azure, follow the `initial steps
+for signup`_ while selecting the CrateDB Cloud Contract as your subscription
+plan. This will automatically alert our Sales team, who will get in touch with
+you to configure the specifics of your contract according to your needs.
+
+
+AWS
+---
+
+To sign a CrateDB Cloud Contract via AWS, simply follow the `link`_ to the
```
Comment (mechanomi) [inline docs/subscription-plans.rst:94]: as a general rule of thumb, it is better to phrase hyperlinks so that the link text itself functions as a noun phrase

so in this case, I would reword this like so:

> To sign a CrateDB Cloud Contract via AWS, visit the [contract page](http://example.com/ on the AWS Marketplace.

a good way to think about this is to consider the fact that some screenreaders (web browsers designed for blind and visually impaired visitors) construct a separate list of all links in the document as a sort of erzatz navigation menu

to imagine what this is like, take every link and add the link text (by itself) to a bullet point list. then consider whether that those links will make sense devoid of their surrounding context

so, for example, a portion of the link menu for this document might read:

* Azure Marketplace
* AWS Marketplace
* Sales team (mailto)
* initial steps for signup
* link
* Azure offer page
* AWS offer page

in this instance, "link" does not indicate to the visitor where the link will take them, so it is a good candidate for being rephrased

I will add a note about this to our style guide

**Review by mechanomi — CHANGES_REQUESTED** [review]
left one comment. otherwise, lgtm

File: docs/glossary.rst [hunk 2]
```diff
@@ -313,8 +313,9 @@ towards particular customer use cases: lower capacity vs. higher capacity, more
 storage vs. more processing power, and so forth. They can also be further
 adjusted for different :ref:`scale units<glossary-scale-unit>` per plan.
 Currently there are four subscription plans available on our cloud offers, with
-more to come in the near future. For more information on choosing the right
-subscription plan, refer to our documentation `on the subject`_.
+more to come in the near future, as well as a contract offer. For more
+information on choosing the right subscription plan, refer to our documentation
+`on the subject`_.
```
Comment (mechanomi) [inline docs/glossary.rst:318]: oops sorry just caught this, but "on the subject" is another candidate for being reworded as a noun phrase :)

File: docs/subscription-plans.rst (hunk 1 above)
Comment (matthijskrul) [reply docs/subscription-plans.rst:94]: Aha, I did not know that! That is an excellent point

**Review by mechanomi — APPROVED** [review]
